src/graphdb.js

```diff
@@ -52,20 +52,67 @@ function toPascalCase (str) {
     return r.trim();
 }
 
+// Changes every instance of invalid characters in the given label with the following delimiters
+function replaceCleanseLabel(label) {
+    let delimiterColon = "_colon_";
+    let delimiterPeriod = "_period_";
+    let delimiterHyphen = "_hyphen_";
+
+    return label
+        .replaceAll(":", delimiterColon)
+        .replaceAll(".", delimiterPeriod)
+        .replaceAll("-", delimiterHyphen);
+}
+
+// Replaces invalid characters in node.label, property.name, edge.label, direction.from, and direction.to
+function replaceCleanseSchema(schema) {
+    schema.nodeStructures.forEach( node => {
+        node.label = replaceCleanseLabel(node.label);
+        node.properties.forEach( property => {
+            property.name = replaceCleanseLabel(property.name);
+        });
+    });
+
+    schema.edgeStructures.forEach( edge => {
+        edge.label = replaceCleanseLabel(edge.label);
+        edge.directions.forEach(direction => {
+            direction.from = replaceCleanseLabel(direction.from);
+            direction.to = replaceCleanseLabel(direction.to);
+        });
+    });
+    
+    return schema;
+}
+
+function checkInvalidChar(label) {
+    let identify = /[:.-]/;
+    return identify.test(label);
+}
+
+function addBackticks(text) {
+    text = `\`${text}\``;
+    return text;
+}
+
 
 function graphDBInferenceSchema (graphbSchema, addMutations) {
     let r = '';
+    let invalidNode = false;
     const gdbs = JSON.parse(graphbSchema);
 
     checkForDuplicateNames(gdbs);
 
     gdbs.nodeStructures.forEach(node => {
         // node type
-        let nodeCase = node.label;
-        if (changeCase) {
-            nodeCase = toPascalCase(node.label); 
+        invalidNode = checkInvalidChar(node.label);
+        let nodeCase = invalidNode ? replaceCleanseLabel(node.label) : node.label;
```

Comment:
Same as [comment](https://github.com/Bit-Quill/amazon-neptune-for-graphql/pull/7/files#r1828501646)

You can just do `let nodeCase = replaceCleanseLabel(node.label);`